xcat-inventory/xcclient/inventory/inventorydiff.py

```diff
@@ -157,48 +156,107 @@ def get_diff_string(self, diff_dict):
 
 class InventoryDiff(object):
 
-    def __init__(self, obj1, obj2, srctype='other', objtype='f'):
```

Comment:
i think the InventoryDiff should better only process the comparison of 2 given dicts, the operations like object export and file load should be abstracted to the high level code logic , so that it can be reused well